Common/src/Storage/TaskLifeCycleHelper.cs

```diff
@@ -431,6 +432,7 @@ await taskTable.RemoveRemainingDataDependenciesAsync(dependentTasks,
 
     await EnqueueReadyTasks(taskTable,
                             pushQueueStorage,
+                            sessionData,
                             readyTasks,
                             cancellationToken)
       .ConfigureAwait(false);
```

Comment:
Instead of checking if the session is paused within this function, I would check before calling the function.

Reply:
The function do other things that we want to keep doing so we need to pass the info to the function